framework/src/engine/endpoint/txpool.ts

```diff
@@ -96,8 +97,17 @@ export class TxpoolEndpoint {
 	}
 
 	// eslint-disable-next-line @typescript-eslint/require-await
-	public async getTransactionsFromPool(_context: RequestContext): Promise<string[]> {
-		return this._pool.getAll().map(tx => tx.getBytes().toString('hex'));
+	public async getTransactionsFromPool(
+		_context: RequestContext,
+	): Promise<JSONObject<TransactionObject>[]> {
+		return this._pool.getAll().map(transaction => ({
```

Comment:
I think we can simply return `TransactionJSON[]` here, and you can just switch to `tx.getJSON()`

Reply:
ah yes...
we can for now dow `(this._pool.getAll() as Transaction[]).map(tx => tx.toJSON())` or add `toJSON` in the interface in the pool